OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1553,6 +1553,32 @@ public static DreamValue NativeProc_length_char(NativeProc.Bundle bundle, DreamO
         return _length(value, false);
     }
 
+    [DreamProc("lerp")]
+    [DreamProcParameter("A")]
+    [DreamProcParameter("B")]
+    [DreamProcParameter("factor", Type = DreamValueTypeFlag.Float)]
+    public static DreamValue NativeProc_lerp(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
+        DreamValue valA = bundle.GetArgument(0, "A");
+        DreamValue valB = bundle.GetArgument(1, "B");
+        DreamValue valFactor = bundle.GetArgument(2, "factor");
+
+        if (!valFactor.TryGetValueAsFloatCoerceNull(out var factor))
+            throw new Exception($"lerp factor {valFactor} is not a num");
+
+        if (valA == DreamValue.Null) valA = new DreamValue(0);
+
+        // TODO: Support non-num arguments like vectors
+        switch (valA.Type) {
+            case DreamValueType.Float when valB.TryGetValueAsFloatCoerceNull(out var floatB): {
```

Comment:
As the implementation of this grows, the switch won't be a good way to determine the argument types. This proc takes matrices, turfs, and vectors which are all different types of DreamObject. This would best be done like:

```dm
if (valA.TryGetValueAsFloatCoerceNull(out var floatA) && valB.TryGetValueAsFloatCoerceNull(out var floatB)) {
    ...
} else if (valA.TryGetValueAsDreamObject<DreamObjectMatrix>(out var matrixA) && valB.TryGetValueAsDreamObject<DreamObjectMatrix>(out var matrixB)) {
    ...
}
```